### ISSUE TITLE
v2: Prevent Fatal error when root package installed as none version published

### ISSUE THREAD

**Issue by llaville** [llaville/box-manifest#5]
# Feature issue

## Summary
Package installed but without public version published throws a PHP Fatal error

## Expected behaviour

Build the PHAR artifact without error

## Actual behaviour
```
PHP Fatal error:  Uncaught TypeError: substr(): Argument #1 ($string) must be of type string, null given
```

## Additional info

Contents of  `composer.json`
```json
{
    "name": "bartlett/sandboxes",
    "description": "For testing purpose only",
    "type": "project",
    "license": "MIT",
    "autoload": {
        "psr-4": {
            "Bartlett\\Sandboxes\\": "src/"
        }
    },
    "authors": [
        {
            "name": "Laurent Laville"
        }
    ],
    "require": {}
}
```

Contents of  `composer.lock`
```
{
    "_readme": [
        "This file locks the dependencies of your project to a known state",
        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
        "This file is @generated automatically"
    ],
    "content-hash": "b7e1016dc3dc8dbd4f23e7f27953509a",
    "packages": [],
    "packages-dev": [],
    "aliases": [],
    "minimum-stability": "stable",
    "stability-flags": [],
    "prefer-stable": false,
    "prefer-lowest": false,
    "platform": [],
    "platform-dev": [],
    "plugin-api-version": "2.3.0"
}
```

Contents of  `vendor/composer/installed.php`
```
<?php return array(
    'root' => array(
        'name' => 'bartlett/sandboxes',
        'pretty_version' => '1.0.0+no-version-set',
        'version' => '1.0.0.0',
        'reference' => NULL,
        'type' => 'project',
        'install_path' => __DIR__ . '/../../',
        'aliases' => array(),
        'dev' => true,
    ),
    'versions' => array(
        'bartlett/sandboxes' => array(
            'pretty_version' => '1.0.0+no-version-set',
            'version' => '1.0.0.0',
            'reference' => NULL,
            'type' => 'project',
            'install_path' => __DIR__ . '/../../',
            'aliases' => array(),
            'dev_requirement' => false,
        ),
    ),
);
```

**Comment by llaville** [comment]
To reproduce : 

```shell
export COMPOSER_ROOT_VERSION=1.0.0+no-version-set && composer update
```

**Comment by llaville** [comment]
Definitively fixed with additional commit 47fdfd9b2bb25056295f70191d4b9e806060c59e

**Comment by llaville** [comment]
Only `SimpleTextManifestBuilder` was fixed, but there is also an impact on `DecorateTextManifestBuilder`
